modules/apps/user/user-test/src/testIntegration/java/com/liferay/user/service/test/UserLocalServiceTest.java

```diff
@@ -864,4 +915,56 @@ private long[] _addUsers(int numberOfUsers) throws Exception {
 	private UserNotificationEventLocalService
 		_userNotificationEventLocalService;
 
+	private class TestAuditMessageProcessor implements AuditMessageProcessor {
+
+		@Override
+		public void process(AuditMessage auditMessage) {
+			if (_eventType != null) {
+				Assert.assertNotNull(auditMessage);
+
+				if ("LOGIN_DNE".equalsIgnoreCase(_eventType)) {
+					Assert.assertTrue(
+						"LOGIN_DNE".equalsIgnoreCase(
+							auditMessage.getEventType()));
+
+					JSONObject additionalInfoJSONObject =
+						auditMessage.getAdditionalInfo();
+
+					String reason = String.valueOf(
+						additionalInfoJSONObject.get("reason"));
+
+					Assert.assertTrue(
+						"Failed to authenticate - User Does Not Exist".
+							equalsIgnoreCase(reason));
+
+					String authType = String.valueOf(
+						additionalInfoJSONObject.get("authType"));
+
+					Assert.assertTrue(
+						"emailAddress".equalsIgnoreCase(authType));
+				}
+			}
```

Comment:
Forgot to mention that I think we can change `equalsIgnoreCase` to `equals` because these are codes that we are making logic decisions on so they should be case sensitive. Unless you found otherwise?

Reply:
Yes, it should be `equals`!
Less indentation will make it more readable. I initially went with this solution because I was thinking whether we want to add assertions for `LOGIN_FAILURE` or `SUCCESS` as well. But it can be changed later if needed.